internal/service/ipam/api_address_block_resource_test.go

```diff
@@ -0,0 +1,1126 @@
+	var resourceName = "bloxone_ipam_address_block.test_ddns_use_conflict_resolution"
+	var v ipam.IpamsvcAddressBlock
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read
+			{
+				Config: testAccAddressBlockDdnsUseConflictResolution("192.168.0.0", "16", "true"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAddressBlockExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "ddns_use_conflict_resolution", "true"),
+				),
+			},
+			// Update and Read
+			{
+				Config: testAccAddressBlockDdnsUseConflictResolution("192.168.0.0", "16", "false"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAddressBlockExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "ddns_use_conflict_resolution", "false"),
+				),
+			},
+			// Delete testing automatically occurs in TestCase
+		},
+	})
+}
+
+func TestAccAddressBlockResource_DhcpConfig(t *testing.T) {
+	var resourceName = "bloxone_ipam_address_block.test_dhcp_config"
+	var v ipam.IpamsvcAddressBlock
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read
+			{
+				Config: testAccAddressBlockDhcpConfig("192.168.0.0", "16", true, true, true, 50, 60),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAddressBlockExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "dhcp_config.allow_unknown", "true"),
+					resource.TestCheckResourceAttr(resourceName, "dhcp_config.allow_unknown_v6", "true"),
+					resource.TestCheckResourceAttr(resourceName, "dhcp_config.ignore_client_uid", "true"),
+					resource.TestCheckResourceAttr(resourceName, "dhcp_config.lease_time", "50"),
+					resource.TestCheckResourceAttr(resourceName, "dhcp_config.lease_time_v6", "60"),
+				),
+			},
+			// Update and Read
+			{
+				Config: testAccAddressBlockDhcpConfig("192.168.0.0", "16", false, true, false, 150, 160),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAddressBlockExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "dhcp_config.allow_unknown", "false"),
+					resource.TestCheckResourceAttr(resourceName, "dhcp_config.allow_unknown_v6", "true"),
+					resource.TestCheckResourceAttr(resourceName, "dhcp_config.ignore_client_uid", "false"),
+					resource.TestCheckResourceAttr(resourceName, "dhcp_config.lease_time", "150"),
+					resource.TestCheckResourceAttr(resourceName, "dhcp_config.lease_time_v6", "160"),
+				),
+			},
+			// Delete testing automatically occurs in TestCase
+		},
+	})
+}
+
+// Test to be enabled once DhcpOptions is implemented
```

Comment:
You can delete this commented code and add a todo instead, e.g 

https://github.com/infobloxopen/terraform-provider-bloxone/blob/35d0e1d2ace37835ce215edd00b8371606f41499/internal/service/ipam/api_subnet_resource_test.go#L18-L26